### PR TITLE
Feature/pre publish filter

### PR DESCRIPTION
**Pull request by CarlHembrough**
### What

* Update s3cypto lib allowing files to be decrypted from S3
* Add test for creating a filter job before a version is published
* Replace `os.Exit` calls with `t.FailNow`, allowing the test output to be written to console when a test fails
* Remove nest 'conveys' as they were adding no value, and cannot be put alongside each other

(note the test currently fails because of keys being overwritten in vault)

### How to review

Review changes / test

### Who can review

Anyone

### CODE REVIEW

File: endToEndTests/policy.hcl [hunk 1]
```diff
@@ -1,3 +1,3 @@
 path "secret/shared/psk" {
-  capabilities = ["read", "create", "update"]
+  capabilities = ["read", "create", "update", "list"]
```
Comment (mattrout92) [inline endToEndTests/policy.hcl]: Why do the e2e tests need to list?

File: endToEndTests/setup.go [hunk 2]
```diff
@@ -152,6 +162,11 @@ func getS3File(region, bucket, filename string, decrypt bool) (io.ReadCloser, er
 			return nil, err
 		}
 
+		log.Debug("reading file with psk",
+			log.Data{"vault_path": cfg.VaultPath,
+				"filename": filename,
+				"psk": psk})
```
Comment (mattrout92) [inline endToEndTests/setup.go]: Could you change this to be: 


``` go
"psk": hex.DecodeString(psk)
```

 just to make it more readable?

**Review by mattrout92 — APPROVED** [review]

File: endToEndTests/policy.hcl (hunk 1 above)
Comment (CarlHembrough) [reply endToEndTests/policy.hcl]: They don't 😀 - I have removed it

File: endToEndTests/setup.go (hunk 2 above)
Comment (CarlHembrough) [reply endToEndTests/setup.go]: Done

**Pull request merged**